server/src/main/java/org/elasticsearch/action/admin/cluster/stats/CCSUsageTelemetry.java

```diff
@@ -129,8 +151,12 @@ private synchronized void doUpdate(CCSUsage ccsUsage) {
             ccsUsage.getSkippedRemotes().forEach(remote -> byRemoteCluster.computeIfAbsent(remote, PerClusterCCSTelemetry::new).skipped());
         }
         ccsUsage.getFeatures().forEach(f -> featureCounts.computeIfAbsent(f, k -> new LongAdder()).increment());
-        if (ccsUsage.getClient() != null) {
-            clientCounts.computeIfAbsent(ccsUsage.getClient(), k -> new LongAdder()).increment();
+        String client = ccsUsage.getClient();
+        if (client != null) {
+            // We only keep limited number of unknown clients to prevent memory exhaustion
+            if (KNOWN_CLIENTS.contains(client) || clientCounts.containsKey(client) || clientCounts.size() < MAX_CLIENTS) {
```

Comment:
My vote is to not do it this way. I would just only record KNOWN_CLIENTS, not allow unknown. The key goal from Product is to know how many queries are coming from Kibana and if the first 20 in the door are some random client that is not kibana (or one of our clients), this will not allow us to get the info we need.

Reply:
OK, just recording known clients is even easier, the only problem is we may miss some, but if we're actually interested only in known ones and mainly Kibana then we could just use known ones.

Reply:
OK, I'd say let's change it now and if product wants more later we can add additional logic.